nflow-rest-api-spring-web/src/main/java/io/nflow/rest/v1/springweb/WorkflowInstanceResource.java

```diff
@@ -57,25 +65,32 @@
   private final CreateWorkflowConverter createWorkflowConverter;
   private final ListWorkflowInstanceConverter listWorkflowConverter;
   private final WorkflowInstanceFactory workflowInstanceFactory;
+  private final WorkflowInstanceDao workflowInstanceDao;
 
   @Inject
   public WorkflowInstanceResource(WorkflowInstanceService workflowInstances, CreateWorkflowConverter createWorkflowConverter,
-      ListWorkflowInstanceConverter listWorkflowConverter, WorkflowInstanceFactory workflowInstanceFactory) {
+      ListWorkflowInstanceConverter listWorkflowConverter, WorkflowInstanceFactory workflowInstanceFactory,
+      WorkflowInstanceDao workflowInstanceDao) {
     this.workflowInstances = workflowInstances;
     this.createWorkflowConverter = createWorkflowConverter;
     this.listWorkflowConverter = listWorkflowConverter;
     this.workflowInstanceFactory = workflowInstanceFactory;
+    this.workflowInstanceDao = workflowInstanceDao;
   }
 
   @PutMapping(consumes = APPLICATION_JSON_VALUE)
   @ApiOperation(value = "Submit new workflow instance")
   @ApiResponses(@ApiResponse(code = 201, message = "Workflow was created", response = CreateWorkflowInstanceResponse.class))
-  public ResponseEntity<CreateWorkflowInstanceResponse> createWorkflowInstance(
+  public ResponseEntity<?> createWorkflowInstance(
       @RequestBody @ApiParam(value = "Submitted workflow instance information", required = true) CreateWorkflowInstanceRequest req) {
     WorkflowInstance instance = createWorkflowConverter.convert(req);
-    long id = workflowInstances.insertWorkflowInstance(instance);
-    instance = workflowInstances.getWorkflowInstance(id, EnumSet.of(WorkflowInstanceInclude.CURRENT_STATE_VARIABLES), null);
-    return ResponseEntity.created(URI.create(String.valueOf(id))).body(createWorkflowConverter.convert(instance));
+    try {
+      long id = workflowInstances.insertWorkflowInstance(instance);
+      instance = workflowInstances.getWorkflowInstance(id, EnumSet.of(WorkflowInstanceInclude.CURRENT_STATE_VARIABLES), null);
+      return created(URI.create(String.valueOf(id))).body(createWorkflowConverter.convert(instance));
+    } catch (StateVariableValueTooLongException e) {
```

Comment:
Also here. If you let the `StateVariableValueTooLongException` be thrown but add exception mapper you can keep the return type as `CreateWorkflowInstanceResponse`

Reply:
Fixed.